devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy

```diff
@@ -0,0 +1,585 @@
+			break
+		default:
+			log.debug "Unknown Sensor Type: ${cmd.sensorType}"
+	}
+	return []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	log.trace "ConfigurationReport ${cmd}"
+
+	runIn(4, refreshPendingChanges)
+
+	def param = configParams.find { it.num == cmd.parameterNumber }
+	if (param) {
+		def val = cmd.scaledConfigurationValue
+
+		log.debug "${param.name}(#${param.num}) = ${val}"
+		state["configParam${param.num}"] = val
+	} else {
+		log.debug "Parameter #${cmd.parameterNumber} = ${cmd.configurationValue}"
+	}
+	return []
+}
+
+def refreshPendingChanges() {
+	sendEvent(name: "pendingChanges", value: "${pendingChanges} Pending Changes", displayed: false)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	log.trace "NotificationReport: $cmd"
+	def result = []
+
+	if(cmd.notificationType == NOTIFICATION_TYPE_ACCESS_CONTROL){
+		if(cmd.event == NOTIFICATION_EVENT_DOOR_WINDOW_OPEN){
+			result << sensorValueEvent(1)
+		} else if(cmd.event == NOTIFICATION_EVENT_DOOR_WINDOW_CLOSED) {
+			result << sensorValueEvent(0)
+		}
+	} else if (cmd.notificationType == NOTIFICATION_TYPE_HOME_SECURITY) {
+		if (cmd.event == NOTIFICATION_EVENT_STATE_IDLE) {
+			result << createEvent(descriptionText: "$device.displayName covering was restored", isStateChange: true)
+			cmds = [zwave.batteryV1.batteryGet(), zwave.wakeUpV1.wakeUpNoMoreInformation()]
+			result << response(commands(cmds, 1000))
+		} else if (cmd.event == NOTIFICATION_EVENT_INSTRUSION_WITH_LOCATION || cmd.event == NOTIFICATION_EVENT_INSTRUSION) {
+			result << sensorValueEvent(1)
+		} else if (cmd.event == NOTIFICATION_EVENT_TEMPERING) {
+			result << createEvent(descriptionText: "$device.displayName covering was removed", isStateChange: true)
+		}
+	} else if (cmd.notificationType) {
+		def text = "Notification $cmd.notificationType: event ${([cmd.event] + cmd.eventParameter).join(", ")}"
+		result << createEvent(name: "notification$cmd.notificationType", value: "$cmd.event", descriptionText: text, displayed: false)
+	} else {
+		def value = cmd.v1AlarmLevel == 255 ? "active" : cmd.v1AlarmLevel ?: "inactive"
+		result << createEvent(name: "alarm $cmd.v1AlarmType", value: value, displayed: false)
+	}
+
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensorbinaryv2.SensorBinaryReport cmd) {
+	log.trace "SensorBinaryReport: $cmd"
+	def map = [:]
+	map.value = cmd.sensorValue ? "open" : "closed"
+	map.name = "contact"
+	if (map.value == "open") {
+		map.descriptionText = "${device.displayName} is open"
+	} else {
+		map.descriptionText = "${device.displayName} is closed"
+	}
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.indicatorv1.IndicatorReport cmd) {
+	log.trace "${cmd}"
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.debug "Ignored Command: $cmd"
+	return []
+}
+
+private getEventMap(name, value, displayed=null, desc=null, unit=null) {
```

Comment:
were the defaults filled in by passing a map directly to `sendEvent` not sufficient?

Reply:
So far, in the three places where this function is applied, the test has not found any problems, so the defaults filled in by passing a map directly to sendEvent should be sufficient.